### PR TITLE
- added Action classes for resize

### PR DESCRIPTION
**Pull request by aschnell**

### CODE REVIEW

File: storage/Devices/BlkDeviceImpl.cc [hunk 1]
```diff
@@ -99,6 +99,53 @@ namespace storage
...
+
+	    // TODO handle children that cannot be resized
+
+	    try
+	    {
+		const Filesystem* filesystem = get_filesystem();
+		actions.push_back(new Action::Resize(filesystem->get_sid(), ResizeMode::SHRINK));
+	    }
+	    catch (const Exception&)
+	    {
```
Comment (shundhammer) [inline storage/Devices/BlkDeviceImpl.cc]: Missing // TO DO

File: storage/Devices/BlkDeviceImpl.cc [hunk 2]
```diff
@@ -99,6 +99,53 @@ namespace storage
...
+
+	    // TODO handle children that cannot be resized
+
+	    try
+	    {
+		const Filesystem* filesystem = get_filesystem();
+		actions.push_back(new Action::Resize(filesystem->get_sid(), ResizeMode::GROW));
+	    }
+	    catch (const Exception&)
+	    {
```
Comment (shundhammer) [inline storage/Devices/BlkDeviceImpl.cc]: Missing // TO DO

**Comment by shundhammer** [comment]
LGTM so far.

What I don't see, though, is a function to query a partition how small it can be shrunk or how large it can be resized. I guess this will come in the near future?

**Comment by aschnell** [comment]
Yes, functionality to query the minimal and maximal possible sizes will follow.

**Pull request merged**